projects/fal/src/fal/app.py

```diff
@@ -232,13 +232,29 @@ async def mirror_input(queue: deque[Any], websocket: WebSocket) -> None:
             queue.append(input)
 
     async def mirror_output(self, queue: deque[Any], websocket: WebSocket) -> None:
+        loop = asyncio.get_event_loop()
+        outgoing_messages: asyncio.Queue[bytes] = asyncio.Queue(maxsize=buffering or 1)
+
+        async def emit(message):
+            await websocket.send_bytes(message)
+
+        async def background_emitter():
+            while True:
+                output = await outgoing_messages.get()
+                await emit(output)
+
+        emitter = asyncio.create_task(background_emitter())
+
         while True:
             if not queue:
                 await asyncio.sleep(0.05)
                 continue
 
             input = queue.popleft()
-            if input is None:
+            if input is None or emitter.done():
+                emitter.cancel()
```

Comment:
This seems funny to me like the more idiomatic way would be to raise a `StopIteration` exception or something. Not a blocking concern for me though

Reply:
raising a StopIteration in where?

Reply:
It's really a redesign to use that since it's more idiomatic

Reply:
ah, you mean the queue not EOF'ing with none but instead we raising a StopIteration inside the loop. I dont really touch that part of the code but i get the idea 👍🏻